File: frontend/src/types/disease-pressure.ts
export interface DiseasePressure {
  id: number;
  date: string;
  location_id: number;
  disease: string;
  risk_score: number | null;
  created_at?: string;
  type: string; // 'historical' or 'forecast'
  is_forecast?: boolean;
}

export interface DiseasePressureList {
  id: number;
  date: string;
  location_id: number;
  disease: string;
  risk_score: number | null;
  type: string; // 'historical' or 'forecast'
  is_forecast?: boolean;
}

export type DiseaseType = "smith_kerns" | "general";

export interface DiseasePressureFilters {
  start_date?: string;
  end_date?: string;
  disease?: DiseaseType;
}

File: backend/app/schemas/disease_pressure.py
from pydantic import BaseModel
from datetime import date, datetime
from typing import Optional


class DiseasePressureRead(BaseModel):
    id: int
    date: date
    location_id: int
    disease: str
    risk_score: Optional[float] = None
    created_at: Optional[datetime] = None
    is_forecast: Optional[bool] = None

    class Config:
        from_attributes = True


class DiseasePressureList(BaseModel):
    id: int
    date: date
    location_id: int
    disease: str
    risk_score: Optional[float] = None
    is_forecast: Optional[bool] = None

    class Config:
        from_attributes = True
